packages/compass-query-bar/src/components/query-bar.tsx

```diff
@@ -350,6 +353,7 @@ export default connect(
         if (applied === false) {
           return;
         }
+        dispatch(fetchSavedQueries());
```

Comment:
This is a bit unconventional for redux to spread the logic across UI and actions like that. If what we're trying to achieve here is "when applying query, saved queries in state should be updated", this logic should be part of the same thunk action, in this case it should be part of the applyQuery.

In your particular case, because `applyQuery` already dispatches `saveRecentQuery` and you added logic to dispatch query loading to the `saveRecentQuery` action, this is redundant and should be just removed